### DISCUSSION TITLE
Installation Error on Clean Ubuntu 20.04 LTS

### DISCUSSION THREAD

**Discussion by AlvaroFFF** [apache/superset#18317] [Q&A / Help]
Hi all,

When trying to install Apache Superset on a clear Ubuntu 20.04 virtual machine I simply follow these steps: 

1. sudo apt-get install build-essential libssl-dev libffi-dev python3-dev python3-pip libsasl2-dev libldap2-dev
2. pip install virtualenv
3. python3 -m venv superset -> Fails and suggests me to execute "apt install python3.8-venv"
4. sudo apt install python3.8-venv
5. python3 -m venv superset
6. . superset/bin/activate
7. pip install apache-superset

Installation ends with errors:

Building wheels for collected packages: apache-superset, cron-descriptor, holidays, pgsanity, python-geohash, sqlalchemy-utils, wtforms-json, Flask-JWT-Extended, Flask-Login, Flask-OpenID, pymeeus
  Building wheel for apache-superset (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /home/alvaro/superset/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-9cjmkhn2/apache-superset/setup.py'"'"'; __file__='"'"'/tmp/pip-install-9cjmkhn2/apache-superset/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-6if0c298
       cwd: /tmp/pip-install-9cjmkhn2/apache-superset/
  Complete output (10 lines):
  -==--==--==--==--==--==--==--==--==--==--==--==--==--==--==-
  VERSION: 1.2.0
  GIT SHA:
  -==--==--==--==--==--==--==--==--==--==--==--==--==--==--==-
  usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: setup.py --help [cmd1 cmd2 ...]
     or: setup.py --help-commands
     or: setup.py cmd --help
  
  error: invalid command 'bdist_wheel'
  ----------------------------------------
  ERROR: Failed building wheel for apache-superset
  Running setup.py clean for apache-superset
  Building wheel for cron-descriptor (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /home/alvaro/superset/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-9cjmkhn2/cron-descriptor/setup.py'"'"'; __file__='"'"'/tmp/pip-install-9cjmkhn2/cron-descriptor/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-pssje3qe
       cwd: /tmp/pip-install-9cjmkhn2/cron-descriptor/
  Complete output (6 lines):
  usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: setup.py --help [cmd1 cmd2 ...]
     or: setup.py --help-commands
     or: setup.py cmd --help
  
  error: invalid command 'bdist_wheel'
  ----------------------------------------
  ERROR: Failed building wheel for cron-descriptor
  Running setup.py clean for cron-descriptor
  Building wheel for holidays (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /home/alvaro/superset/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-9cjmkhn2/holidays/setup.py'"'"'; __file__='"'"'/tmp/pip-install-9cjmkhn2/holidays/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-cuyj6ksy
       cwd: /tmp/pip-install-9cjmkhn2/holidays/
  Complete output (6 lines):
  usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: setup.py --help [cmd1 cmd2 ...]
     or: setup.py --help-commands
     or: setup.py cmd --help
  
  error: invalid command 'bdist_wheel'
  ----------------------------------------

(Error follows)

And if I try to run "superset db upgrade" returns error as well:

(superset) alvaro@alvaro-VirtualBox:~$ superset db upgrade
Traceback (most recent call last):
  File "/home/alvaro/superset/lib/python3.8/site-packages/pkg_resources/__init__.py", line 583, in _build_master
    ws.require(__requires__)
  File "/home/alvaro/superset/lib/python3.8/site-packages/pkg_resources/__init__.py", line 900, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/home/alvaro/superset/lib/python3.8/site-packages/pkg_resources/__init__.py", line 791, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (itsdangerous 2.0.1 (/home/alvaro/superset/lib/python3.8/site-packages), Requirement.parse('itsdangerous<2.0,>=0.24'), {'flask'})

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/alvaro/superset/bin/superset", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/home/alvaro/superset/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3252, in <module>
    def _initialize_master_working_set():
  File "/home/alvaro/superset/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3235, in _call_aside
    f(*args, **kwargs)
  File "/home/alvaro/superset/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3264, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/home/alvaro/superset/lib/python3.8/site-packages/pkg_resources/__init__.py", line 585, in _build_master
    return cls._build_from_requirements(__requires__)
  File "/home/alvaro/superset/lib/python3.8/site-packages/pkg_resources/__init__.py", line 598, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
  File "/home/alvaro/superset/lib/python3.8/site-packages/pkg_resources/__init__.py", line 786, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'itsdangerous<2.0,>=0.24' distribution was not found and is required by flask

I have tried several solutions but none of them worked for me. I have repeated the process in various clean Ubuntu 20.04 virtual machines but is not possible for me to install Superset.

I attach both errors expecting it helps.

[db_init.log](https://github.com/apache/superset/files/6855320/db_init.log)
[install.log](https://github.com/apache/superset/files/6855321/install.log)

### Environment

(please complete the following information):

- superset version: command not working
- python version: 3.8.10
- node.js version: v10.19.0

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [ ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [X] I have reproduced the issue with at least the latest released version of superset.
- [X] I have checked the issue tracker for the same issue and I haven't found one similar.

**Comment by etr2460** [comment]
@AlvaroFFF or @bzmatteo72, we recently released superset 1.3, and wouldn't expect to see this issue anymore with it. Can you confirm if your issue is fixed in superset 1.3? thanks!

**Comment by AlvaroFFF** [comment]
Hi @etr2460,

Issue is still there. Let me try to provide more information about enviroment. These are the steps I've followed:

Created a new machine on Google Cloud Platform, with following specifications:

- Region: europe-west1 (Belgium)
- Zone: europe-west1-b
- Machine type: e2-standard-4 (4 vCPUs, 16GB memory)
- Boot disk: Ubuntu 20.04 LTS, 50GB

Once the instance is up, we start executing commands as mentioned [here](https://superset.apache.org/docs/installation/installing-superset-from-scratch)

`
sudo apt-get install build-essential libssl-dev libffi-dev python3-dev python3-pip libsasl2-dev libldap2-dev
pip install virtualenv
`

After this command we go slightly "out of the guide" to install python venv:

`sudo apt-get install python3.8-venv`

And now we can create and activate our virtual enviroment and follow with the installation:

`python3 -m venv venv`
`. venv/bin/activate`
`pip install apache-superset`

At this point, some errors have appeared in the console while installing superset. Seems like most of them are related with flask version.

Indeed, if we execute `superset --version` an exception is raised.

I have tried to build a workaround that worked to me, consists basically on installing some flask versions manually:

`pip install flask-wtf==0.14.2`
`pip install flask==1.1.0`
`pip install werkzeug==0.16.1`

Once the ¿dependencies error? seems to be fixed we can continue:

`superset db upgrade`
`export FLASK_APP=superset`
`superset fab create-admin`
`superset init`
`superset run -p 8088 --with-threads --reload --debugger`

And it works successfully.

I attach the history of commands hoping it could be useful.
[history.txt](https://github.com/apache/superset/files/7120870/history.txt)

Thank you,

Alvaro

**Comment by etr2460** [comment]
Thanks for the details! Seems like we still need to fix some of our version pinning. Maybe @dpgaspar, @villebro, or @john-bodley could help you out here

**Comment by sakoltsov** [comment]
Hi! Right now I am facing a similar problem, I used the workaround above, but there is one error left due to the markupsafe library. In version 2.1.0(Released 2022-02-17):  
     --Remove _soft_unicode_, which was previously deprecated. Use _soft_str_ instead

Installing the previous version worked for me:
`pip install markupsafe==2.0.0`
`pip install jinja2==3.0.0`

**Reply by Cmagno13** [reply]
(venv) root@ATECHAPek0dexde:/home/cmagno# superset db upgrade
Traceback (most recent call last):
  File "/home/cmagno/venv/bin/superset", line 18, in <module>
    from superset.cli import superset
  File "/home/cmagno/venv/lib/python3.8/site-packages/superset/__init__.py", line 18, in <module>
    from flask import current_app, Flask
  File "/home/cmagno/venv/lib/python3.8/site-packages/flask/__init__.py", line 19, in <module>
    from . import json
  File "/home/cmagno/venv/lib/python3.8/site-packages/flask/json/__init__.py", line 15, in <module>
    from itsdangerous import json as _json
ImportError: cannot import name 'json' from 'itsdangerous' (/home/cmagno/venv/lib/python3.8/site-packages/itsdangerous/__init__.py)

**Comment by SUGAM-ARORA** [comment]
> Hi all,
> 
> When trying to install Apache Superset on a clear Ubuntu 20.04 virtual machine I simply follow these steps:
> 
> 1. sudo apt-get install build-essential libssl-dev libffi-dev python3-dev python3-pip libsasl2-dev libldap2-dev
> 2. pip install virtualenv
> 3. python3 -m venv superset -> Fails and suggests me to execute "apt install python3.8-venv"
> 4. sudo apt install python3.8-venv
> 5. python3 -m venv superset
> 6. . superset/bin/activate
> 7. pip install apache-superset
> 
> Installation ends with errors:
> 
> Building wheels for collected packages: apache-superset, cron-descriptor, holidays, pgsanity, python-geohash, sqlalchemy-utils, wtforms-json, Flask-JWT-Extended, Flask-Login, Flask-OpenID, pymeeus Building wheel for apache-superset (setup.py) ... error ERROR: Command errored out with exit status 1: command: /home/alvaro/superset/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-9cjmkhn2/apache-superset/setup.py'"'"'; **file**='"'"'/tmp/pip-install-9cjmkhn2/apache-superset/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(**file**);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, **file**, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-6if0c298 cwd: /tmp/pip-install-9cjmkhn2/apache-superset/ Complete output (10 lines): -==--==--==--==--==--==--==--==--==--==--==--==--==--==--==- VERSION: 1.2.0 GIT SHA: -==--==--==--==--==--==--==--==--==--==--==--==--==--==--==- usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...] or: setup.py --help [cmd1 cmd2 ...] or: setup.py --help-commands or: setup.py cmd --help
> 
> ## error: invalid command 'bdist_wheel'
> ERROR: Failed building wheel for apache-superset Running setup.py clean for apache-superset Building wheel for cron-descriptor (setup.py) ... error ERROR: Command errored out with exit status 1: command: /home/alvaro/superset/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-9cjmkhn2/cron-descriptor/setup.py'"'"'; **file**='"'"'/tmp/pip-install-9cjmkhn2/cron-descriptor/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(**file**);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, **file**, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-pssje3qe cwd: /tmp/pip-install-9cjmkhn2/cron-descriptor/ Complete output (6 lines): usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...] or: setup.py --help [cmd1 cmd2 ...] or: setup.py --help-commands or: setup.py cmd --help
> 
> ## error: invalid command 'bdist_wheel'
> ERROR: Failed building wheel for cron-descriptor Running setup.py clean for cron-descriptor Building wheel for holidays (setup.py) ... error ERROR: Command errored out with exit status 1: command: /home/alvaro/superset/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-9cjmkhn2/holidays/setup.py'"'"'; **file**='"'"'/tmp/pip-install-9cjmkhn2/holidays/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(**file**);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, **file**, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-cuyj6ksy cwd: /tmp/pip-install-9cjmkhn2/holidays/ Complete output (6 lines): usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...] or: setup.py --help [cmd1 cmd2 ...] or: setup.py --help-commands or: setup.py cmd --help
> 
> ## error: invalid command 'bdist_wheel'
> (Error follows)
> 
> And if I try to run "superset db upgrade" returns error as well:
> 
> (superset) alvaro@alvaro-VirtualBox:~$ superset db upgrade Traceback (most recent call last): File "/home/alvaro/superset/lib/python3.8/site-packages/pkg_resources/**init**.py", line 583, in _build_master ws.require(**requires**) File "/home/alvaro/superset/lib/python3.8/site-packages/pkg_resources/**init**.py", line 900, in require needed = self.resolve(parse_requirements(requirements)) File "/home/alvaro/superset/lib/python3.8/site-packages/pkg_resources/**init**.py", line 791, in resolve raise VersionConflict(dist, req).with_context(dependent_req) pkg_resources.ContextualVersionConflict: (itsdangerous 2.0.1 (/home/alvaro/superset/lib/python3.8/site-packages), Requirement.parse('itsdangerous<2.0,>=0.24'), {'flask'})
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last): File "/home/alvaro/superset/bin/superset", line 6, in from pkg_resources import load_entry_point File "/home/alvaro/superset/lib/python3.8/site-packages/pkg_resources/**init**.py", line 3252, in def _initialize_master_working_set(): File "/home/alvaro/superset/lib/python3.8/site-packages/pkg_resources/**init**.py", line 3235, in _call_aside f(*args, **kwargs) File "/home/alvaro/superset/lib/python3.8/site-packages/pkg_resources/**init**.py", line 3264, in _initialize_master_working_set working_set = WorkingSet._build_master() File "/home/alvaro/superset/lib/python3.8/site-packages/pkg_resources/**init**.py", line 585, in _build_master return cls._build_from_requirements(**requires**) File "/home/alvaro/superset/lib/python3.8/site-packages/pkg_resources/**init**.py", line 598, in _build_from_requirements dists = ws.resolve(reqs, Environment()) File "/home/alvaro/superset/lib/python3.8/site-packages/pkg_resources/**init**.py", line 786, in resolve raise DistributionNotFound(req, requirers) pkg_resources.DistributionNotFound: The 'itsdangerous<2.0,>=0.24' distribution was not found and is required by flask
> 
> I have tried several solutions but none of them worked for me. I have repeated the process in various clean Ubuntu 20.04 virtual machines but is not possible for me to install Superset.
> 
> I attach both errors expecting it helps.
> 
> [db_init.log](https://github.com/apache/superset/files/6855320/db_init.log) [install.log](https://github.com/apache/superset/files/6855321/install.log)
> 
> ### Environment
> (please complete the following information):
> 
> * superset version: command not working
> * python version: 3.8.10
> * node.js version: v10.19.0
> 
> ### Checklist
> Make sure to follow these steps before submitting your issue - thank you!
> 
> * [ ]  I have checked the superset logs for python stacktraces and included it here as text if there are any.
> * [x]  I have reproduced the issue with at least the latest released version of superset.
> * [x]  I have checked the issue tracker for the same issue and I haven't found one similar.

Hello , 
I didn't get any error with these steps but the login page including id and password does not get generated and there error comes which requires us to backtrack .  
Suggestions are highly recommended. Looking forward for...
